### PR TITLE
Add Dino App

### PR DESCRIPTION
**Pull request by chuanchan1116**
Nothing tested yet

### CODE REVIEW

File: fw/Core/Hitcon/App/DinoApp.cc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  _input_buf = BUTTON_NONE;
+  _curr_dino_frame = &dino_running_bitmap[0];
+  _obstacle_interval = 0;
+  _generate_obs_rate = 8;
+  _obstacle_frame_pos = 0;
+  _game_over = 0;
+  _is_dino_down = 0;
+  _dino_ani_frame = 0;
+  _dino_jump_vel = DINO_INITIAL_VEL;
+  scheduler.Queue(&_routine_task, nullptr);
```
Comment (john0312) [inline fw/Core/Hitcon/App/DinoApp.cc:31]: You only need to Queue() once, during Init().

File: fw/Core/Hitcon/Logic/ButtonLogic.h [hunk 2]
```diff
@@ -30,6 +30,7 @@ constexpr int BUTTON_LONG_PRESS_BIT = 1 << 14;
 // Bit 14 - This is a long press if set.
 // Bit 0-3 - The button pressed. See BUTTON_* constants below.
 enum button_t {
+  BUTTON_NONE = 0,
```
Comment (john0312) [inline fw/Core/Hitcon/Logic/ButtonLogic.h:33]: Maybe note that NONE is not a real button, it's just a place holder.

File: fw/Core/Hitcon/App/DinoApp.cc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+          writeObsByte(_obstacle_frame_pos + 2, bird_bitmap[2]);
+          break;
+        case 1:  // big cactuar
+          writeObsByte(_obstacle_frame_pos, big_cactuar_bitmap[0]);
+          writeObsByte(_obstacle_frame_pos + 1, big_cactuar_bitmap[1]);
+          writeObsByte(_obstacle_frame_pos + 2, big_cactuar_bitmap[2]);
+        case 2:  // small tall cactuar
+          writeObsByte(_obstacle_frame_pos, small_tall_cactuar_bitmap[0]);
+        case 3:  // small short cactuar
+          writeObsByte(_obstacle_frame_pos, small_short_cactuar_bitmap[0]);
```
Comment (Lance798) [inline fw/Core/Hitcon/App/DinoApp.cc:73]:
```
break;
_obstacle_interval = 0;
```

File: fw/Core/Hitcon/App/DinoApp.cc [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    _game_over = true;
+    badge_controller.change_app(&main_menu);
+    return;
+  }
+  printFrame();
+  _dino_ani_frame ^= 1;
+}
+
+inline void DinoApp::writeObsByte(uint8_t pos, display_buf_t line) {
+  _obstacle_frame[pos] = line;
```
Comment (Lance798) [inline fw/Core/Hitcon/App/DinoApp.cc:109]: pos += 15;
目前看起來障礙物會直接加到小恐龍身上 就直接game over

Comment (chuanchan1116) [reply fw/Core/Hitcon/App/DinoApp.cc:109]: Ah, it should be multiplied by 2

**Pull request merged**